lib/matplotlib/image.py

```diff
@@ -985,6 +985,15 @@ def set_data(self, x, y, A):
                     self.is_grayscale = True
             else:
                 raise ValueError("3D arrays must have RGB or RGBA as last dim")
+
+        # For efficient cursor readout, ensure x and y are increasing.
+        if x[-1] < x[0]:
```

Comment:
Do we enforce that x/y are monotonic someplace else?

If we bump the minimum numpy to 1.7, `searchsorted` takes in a `sorter` kwarg which is an array with the indexes in sorted order (https://docs.scipy.org/doc/numpy/reference/generated/numpy.searchsorted.html)